### ISSUE TITLE
Cannot create GNNGraph with unconnected nodes

### ISSUE THREAD

**Issue by sandijs-private** [CarloLucibello/GraphNeuralNetworks.jl#472]
```
using GraphNeuralNetworks

g = GNNGraph()

add_nodes(g, 5)

@info "GRAPH WITH UNCONNECTED NODES" g

add_edges(g, [1, 2], [3, 4])

@info "GRAPH WITH SOME CONNECTED NODES" g
```

```
sandijs@fedora:~/Documents/bug$ julia bug.jl 
┌ Info: GRAPH WITH UNCONNECTED NODES
│   g =
│    GNNGraph:
│      num_nodes: 0
└      num_edges: 0
┌ Info: GRAPH WITH SOME CONNECTED NODES
│   g =
│    GNNGraph:
│      num_nodes: 0
└      num_edges: 0
```

**Comment by sandijs-private** [comment]
That Graph -> GNNGraph conversion happens in [traffic prediction graph example](https://github.com/CarloLucibello/GraphNeuralNetworks.jl/blob/master/examples/traffic_prediction.jl)
Theoretically, it could lead to this example not working as expected.
Or working slower.
In my case, lower accuracy.

```
function getdataset()
    metrla = METRLA(; num_timesteps = 3)
    g = metrla[1]
    graph = GNNGraph(g.edge_index; edata = g.edge_data, g.num_nodes)
    features = g.node_data.features
    targets = g.node_data.targets
    train_loader = zip(features[1:2000], targets[1:2000])
    test_loader = zip(features[2001:2288], targets[2001:2288])
    return graph, train_loader, test_loader
end
```

**Comment by CarloLucibello** [comment]
`add_nodes` and `add_edges` are not mutating, so the correct version of the first example would be
```julia
julia> using GraphNeuralNetworks

julia> g = GNNGraph()
GNNGraph:
  num_nodes: 0
  num_edges: 0

julia> g= add_nodes(g, 5)
GNNGraph:
  num_nodes: 5
  num_edges: 0

julia> @info "GRAPH WITH UNCONNECTED NODES" g
┌ Info: GRAPH WITH UNCONNECTED NODES
│   g =
│    GNNGraph:
│      num_nodes: 5
└      num_edges: 0

julia> g= add_edges(g, [1, 2], [3, 4])
GNNGraph:
  num_nodes: 5
  num_edges: 2

julia> @info "GRAPH WITH SOME CONNECTED NODES" g
┌ Info: GRAPH WITH SOME CONNECTED NODES
│   g =
│    GNNGraph:
│      num_nodes: 5
└      num_edges: 2
```

**Comment by CarloLucibello** [comment]
The `Graph` type from `Graphs.jl` is undirected. `GNNGraph`s are directed instead, so when converting from the former to the latter each undirected edge is converted to two directed edges. 

I don't think there are any conversion problems in the `METRLA` example.
